### PR TITLE
Seed is uint64 t

### PR DESCRIPTION
**Pull request by ClementPernet**
The types used for seeds in LinBox vary a lot and cause numerous type conversion ambiguities with Givaro random iterators on OS-X where size_t is not silently converted into an uint64_t: https://ci.inria.fr/linbox/job/LinBox/290/
This PR fixes all such issues by changing all seeds to be uint64_t (excpect the one of Mersenne Twister which is already uint32_t since everything there is on 32 bits).

### CODE REVIEW

File: linbox/ring/ntl/ntl-lzz_p.h [hunk 1]
```diff
@@ -340,10 +340,10 @@ namespace LinBox
 
 		UnparametricRandIter<NTL::zz_p> (const NTL_zz_p & F,
                                          const integer& size=0,
-                                         const integer& seed=0) :
+                                         const uint64_t& seed=0) :
                 _size(size), _seed(seed), _ring(F)
             {
-                if (_seed == integer(0)) _seed = int64_t(time(NULL));
+                if (_seed == 0) _seed = int64_t(time(NULL));
```
Comment (jgdumas) [inline linbox/ring/ntl/ntl-lzz_p.h]: why not uint64_t(time(NULL)) ?
Note, that there is also Givaro::BaseTimer::seed() ...

**Comment by ClementPernet** [comment]
Dictator merge

File: linbox/ring/ntl/ntl-rr.h [hunk 2]
```diff
@@ -280,16 +280,17 @@ namespace LinBox
...
 	public:
 
 		UnparametricRandIter<NTL::RR> (const NTL_RR & F,
 					       const integer& size = 0,
-					       const integer& seed = 0) :
+					       const uint64_t seed = 0) :
                 _size(size), _seed(seed), _ring(F)
 		{
-			if (_seed == integer(0)) _seed = int64_t(time(NULL));
+			if (_seed == 0) _seed = time(NULL);
```
Comment (jgdumas) [inline linbox/ring/ntl/ntl-rr.h]: uint64_t cast

File: linbox/vector/stream.h [hunk 3]
```diff
@@ -437,15 +437,15 @@ namespace LinBox
 		typedef _Vector Vector;
 
 		RandomSparseStream (const Field &F, RandIter &r, double p,
-				    size_t N, size_t M = 0, int seed=(int)time (NULL)) :
+				    size_t N, size_t M = 0, uint64_t seed= time (NULL)) :
```
Comment (jgdumas) [inline linbox/vector/stream.h]: uint64_t cast

File: linbox/vector/stream.h [hunk 4]
```diff
@@ -517,9 +517,9 @@ namespace LinBox
 		typedef RandomSparseStream<Field, Vector, RandIter, VectorCategories::SparseAssociativeVectorTag > Self_t;
 
 		RandomSparseStream (const Field &F, RandIter &r, double p, size_t N,
-				    size_t M = 0, int seed=(int)time (NULL)) :
+				    size_t M = 0, uint64_t seed= time (NULL)) :
```
Comment (jgdumas) [inline linbox/vector/stream.h]: idem

File: linbox/vector/stream.h [hunk 5]
```diff
@@ -571,9 +571,9 @@ namespace LinBox
 		typedef _Vector Vector;
 		typedef RandomSparseStream<Field, Vector, RandIter, VectorCategories::SparseParallelVectorTag > Self_t;
 
-		RandomSparseStream (const Field &F, RandIter &r, double p, size_t nn, size_t mm = 0, int seed=(int)time (NULL)) :
+		RandomSparseStream (const Field &F, RandIter &r, double p, size_t nn, size_t mm = 0, uint64_t  seed= time (NULL)) :
```
Comment (jgdumas) [inline linbox/vector/stream.h]: id

File: linbox/ring/ntl/ntl-rr.h (hunk 2 above)
Comment (ClementPernet) [reply linbox/ring/ntl/ntl-rr.h]: the cast is silently done by the compiler

File: linbox/ring/ntl/ntl-lzz_p.h (hunk 1 above)
Comment (ClementPernet) [reply linbox/ring/ntl/ntl-lzz_p.h]: I did remove that cast

File: linbox/vector/stream.h (hunk 4 above)
Comment (ClementPernet) [reply linbox/vector/stream.h]: idem

**Pull request merged**